### ISSUE TITLE
Alarm "at the time of event" serialized incorrectly

### ISSUE THREAD

**Issue by Jeff-Lowrey** [C4ptainCrunch/ics.py#253]
I've got some events created from the Apple calendar, which among other things let's me specify an alarm to go off "at the time of the event".

In the ICS output, this gets rendered as 
TRIGGER:PT0S

ics.py parses this successfully into an Alarm object (it happens to be an audio alarm).  But it throws an error when it trys to serialize the event back into an ICS format.

It throws a ValueError of "Alarm must have a trigger".  This comes from the serialize_trigger method of the BaseAlarmSerializer. 
```
class BaseAlarmSerializer(Serializer):
    def serialize_trigger(alarm, container):
        if not alarm.trigger:
            raise ValueError("Alarm must have a trigger")
```

The trigger value is set to a valid timedelta. It just happens to be a timedelta with a zero value. (timedelta: 0:00:00)

Should the serializer check to see if the trigger is None instead of whether or not it evaluates to false?

**Comment by N-Coder** [comment]
Yep, using `is None` sounds right, thanks! Could you make a PR against the [version-0.7 branch](https://github.com/C4ptainCrunch/ics.py/tree/version-0.7) so that we can include this in a v0.7.1 release? It would be awesome if you could also scrap any personal data from that event and post it here, so we can use it for regression testing. :smiley:

**Comment by Jeff-Lowrey** [comment]
I've actually been having some issues replicating this against an environment built from a clone of the repo and poetry. It is still repeatable in my IDE-based python runtime.

So I'm not sure I'm the right one to open a PR? 

Here's my best stab at an anonymized version of the event that I'm testing against. As you can see, there's three alarms, one at 15 minutes before the event, one at five, and one at the time of the event. 
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Apple Inc.//Mac OS X 10.15.6//EN
BEGIN:VEVENT
DTEND;TZID=America/Chicago:20200815T100000
TRANSP:OPAQUE
SUMMARY:Lets do something!
DTSTAMP:20200802T154318Z
DTSTART;TZID=America/Chicago:20200815T090000
SEQUENCE:1
BEGIN:VALARM
TRIGGER:-PT15M
ATTACH;VALUE=URI:Chord
ACTION:AUDIO
END:VALARM
BEGIN:VALARM
TRIGGER:-PT5M
ATTACH;VALUE=URI:Chord
ACTION:AUDIO
END:VALARM
BEGIN:VALARM
TRIGGER:PT0S
ATTACH;VALUE=URI:Chord
ACTION:AUDIO
END:VALARM
END:VEVENT
END:VCALENDAR
```

**Comment by N-Coder** [comment]
Sounds like you used the `master` branch (which is already on its way towards the 0.8 version, introducing poetry amongst a lot of other changes) instead of the `version-0.7` branch, which tracks the current stable version. I made a quick PR via the GitHub online editor: #254. Could you try that out? Unfortunately, the test suite is currently failing due to some breaking changes in pytest...

**Comment by Jeff-Lowrey** [comment]
I've tried #234, and it *almost* works.

When I print the parsed calendar out, instead of getting the missing trigger failure, I get the below. Notice the "TRIGGER:P" instead of "TRIGGER:PT0S"

```BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Apple Inc.//Mac OS X 10.15.6//EN
CALSCALE:GEORGIAN
BEGIN:VEVENT
SEQUENCE:1
BEGIN:VALARM
ACTION:AUDIO
ATTACH;VALUE=URI:Chord
TRIGGER:P
END:VALARM
BEGIN:VALARM
ACTION:AUDIO
ATTACH;VALUE=URI:Chord
TRIGGER:-PT5M
END:VALARM
BEGIN:VALARM
ACTION:AUDIO
ATTACH;VALUE=URI:Chord
TRIGGER:-PT15M
END:VALARM
DTSTAMP:20200802T154318Z
DTEND:20200815T150000Z
DTSTART:20200815T140000Z
SUMMARY:Lets do something!
TRANSP:OPAQUE
UID:f73e80a0-7e8b-4329-8338-904fe4963d66@f73e.org
END:VEVENT
END:VCALENDAR
```

**Comment by Jeff-Lowrey** [comment]
If I patch ics.utils.timedelta_to_duration like so
```
    if days:
        res += str(days) + 'D'
    if seconds == 0 :
        res += 'T0S'
    if seconds:
        res += 'T'
```
Then it works.

This might not be the best way to to patch, however.

**Comment by N-Coder** [comment]
I already [fixed](https://github.com/C4ptainCrunch/ics.py/blob/master/src/ics/valuetype/datetime.py#L221) that function on `master`/0.8, I'll backport these changes and include them in the PR once I found time to test that everything works. :+1:

**Comment by Jeff-Lowrey** [comment]
If it's fixed in 0.8, then it doesn't need to be backported - at least for me. I was just playing around with ICS trying to simplify a task of turning a spreadsheet of tasks and dates into an iCalendar file.  But I spent the time to create them manually, so I don't think I'll be using ics-py for another bit.

So this issue can be closed, and you don't have to back-port unless you want to. 

I was using master/0.8 because that's what came when I did 'pip install ics'.
